packages/gatekeeper/src/helpers/oauth.ts

```diff
@@ -64,7 +65,30 @@ const addSecToCurrentTime = (seconds: number) => {
 /** Function to get OpenSRP user info from api response object
  * @param {{[key: string]: any }} apiResponse - the API response object
  */
-export function getOpenSRPUserInfo(apiResponse: { [key: string]: any }): SessionState {
+export function getOpenSRPUserInfo(apiRes: { [key: string]: any }): SessionState {
+  const {
+    email_verified,
+    oAuth2Data,
+    given_name,
+    family_name,
+    preferred_username,
+    realm_access,
+    sub,
+    name,
+    organization
+  } = apiRes;
+  const apiResponse = {
+    roles: (realm_access?.roles ?? []).map((role: string) => `ROLE_${role}`),
+    email: null,
+    username: preferred_username,
+    user_id: sub,
+    preferred_name: name,
+    family_name,
+    given_name,
+    email_verified,
+    oAuth2Data,
+    organization
+  };
```

Comment:
We are moving to only fully work with keycloak. i.e. where before the endpoint used here was `<opensrp-base-url>/user-details`, its now going to be `<keycloak-base-url>/auth/realms/FHIR_Android/protocol/openid-connect/userinfo` .

Reply:
Does this server both OpenSRP web and FHIR web correctly?

Reply:
@dubdabasoduba yeap, it will work with all instances which use keyloack